pkg/sensors/tracing/killer.go

```diff
@@ -79,35 +92,69 @@ func loadMultiKillerSensor(bpfDir, mapDir string, load *program.Program, verbose
 	return nil
 }
 
-func (k *killerSensor) LoadProbe(args sensors.LoadProbeArgs) error {
+func (kh *killerHandler) LoadProbe(args sensors.LoadProbeArgs) error {
+	if args.Load.Label == "kprobe.multi/killer" {
+		return kh.loadMultiKillerSensor(args.BPFDir, args.MapDir, args.Load, args.Verbose)
+	}
 	if args.Load.Label == "kprobe/killer" {
-		return loadSingleKillerSensor(args.BPFDir, args.MapDir, args.Load, args.Verbose)
+		return kh.loadSingleKillerSensor(args.BPFDir, args.MapDir, args.Load, args.Verbose)
+	}
+
+	if strings.HasPrefix(args.Load.Label, "fmod_ret/") {
+		return program.LoadFmodRetProgram(args.BPFDir, args.MapDir, args.Load, args.Verbose)
+	}
+
+	return fmt.Errorf("killer loader: unknown label: %s", args.Load.Label)
+}
+
+// select proper override method based on configuration and spec options
+func selectOverrideMethod(specOpts *specOptions) (OverrideMethod, error) {
+	overrideMethod := specOpts.OverrideMethod
+	switch overrideMethod {
+	case OverrideMethodDefault:
+		// by default, first try OverrideReturn and if this does not work try fmod_ret
+		if bpf.HasOverrideHelper() {
+			overrideMethod = OverrideMethodReturn
+		} else if bpf.HasModifyReturn() && bpf.HasModifyReturnSyscall() {
```

Comment:
killer is just for syscalls, so we could use just bpf.HasModifyReturnSyscall ?

Reply:
yeah, we can do that.
I was thinking that it could be extended in the future to hook into security_ functions but we I'll just use HasModifyReturnSyscall for now.

Reply:
I'll change the tests as well.